testing/test_terminal.py

```diff
@@ -3508,3 +3541,25 @@ def test_session_lifecycle(
         # Session finish - should remove progress.
         plugin.pytest_sessionfinish()
         assert "\x1b]9;4;0;\x1b\\" in mock_file.getvalue()
+
+
+def test_terminal_reporter_write_with_capture(pytester: Pytester) -> None:
```

Comment:
This test seems to pass also in main, so needs some refinement if we want to prevent regression.